src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisTable.tsx

```diff
@@ -0,0 +1,318 @@
+import OncoKBTable from 'app/components/oncokbTable/OncoKBTable';
+import {filterByKeyword} from 'app/shared/utils/Utils';
+import autobind from 'autobind-decorator';
+import {action, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import React from 'react';
+import {Row} from 'react-bootstrap';
+import {
+  getUsageTableColumnDefinition,
+  ToggleValue,
+  UsageRecord,
+  UsageTableColumnKey,
+} from 'app/pages/usageAnalysisPage/UsageAnalysisPage';
+import {UsageToggleGroup} from './UsageToggleGroup';
+import {UsageAnalysisCalendarButton} from 'app/components/calendarButton/UsageAnalysisCalendarButton';
+import {
+  emailHeader,
+  endpointHeader,
+  filterDependentTimeHeader,
+  noPrivateEndpointHeader,
+  usageHeader,
+} from 'app/components/oncokbTable/HeaderConstants';
+import {UserOverviewUsage} from "app/shared/api/generated/API";
+import UsageText from "app/shared/texts/UsageText";
+import {PAGE_ROUTE, TABLE_DAY_FORMAT, TABLE_MONTH_FORMAT} from "app/config/constants";
+import moment from "moment";
+import _ from "lodash";
+import {Link} from "react-router-dom";
+import {SortingRule} from "react-table";
+
+type IUserUsageDetailsTable = {
+  data: UserOverviewUsage[];
+  loadedData: boolean;
+  defaultResourcesType: ToggleValue;
+  defaultTimeType: ToggleValue;
+  defaultPageSize?: number;
+};
+
+@observer
+export default class UsageAnalysisTable extends React.Component<
+  IUserUsageDetailsTable,
+  {}
+  > {
+  @observable resourcesTypeToggleValue: ToggleValue = this.props
+    .defaultResourcesType;
+  @observable timeTypeToggleValue: ToggleValue = this.props.defaultTimeType;
+  @observable fromDate: string | undefined;
+  @observable toDate: string | undefined;
+  @observable filterToggled: boolean;
+  @observable dropdownMenuOpen: boolean;
+
+  @autobind
+  @action
+  handleResourcesTypeToggleChange(value: ToggleValue) {
+    this.resourcesTypeToggleValue = value;
+  }
+
+  @autobind
+  @action
+  handleTimeTypeToggleChange(value: ToggleValue) {
+    this.timeTypeToggleValue = value;
+  }
+
+  @autobind
+  @action
+  handleFilterToggledChange(value: boolean) {
+    this.filterToggled = value;
+  }
+
+  @computed get calculateData(): UserOverviewUsage[] | UsageRecord[] {
```

Comment:
I would rather for you to calculate the data separately and render two tables so the code looks more organized. Attache a `key` to differentiate. Any performance concern here?